bundles/CoreBundle/Command/ThumbnailsImageCommand.php

```diff
@@ -149,54 +169,88 @@ protected function execute(InputInterface $input, OutputInterface $output)
-        );
-        $progress->start();
+                $items[] = serialize($item);
+            }
+        }
+        return $items;
+    }
 
-        for ($i = 0; $i < (ceil($total / $perLoop)); $i++) {
-            $list->setLimit($perLoop);
-            $list->setOffset($i * $perLoop);
-            $images = $list->load();
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function runSingleCommand(string $item, InputInterface $input, OutputInterface $output)
+    {
+        $item = unserialize($item);
 
-            foreach ($images as $image) {
-                if (!$image instanceof Asset\Image) {
-                    continue;
-                }
+        $image = Asset\Image::getById($item['image_id']);
+        if (!$image instanceof Asset\Image) {
+            return;
+        }
 
-                if ($input->getOption('force')) {
-                    foreach ($thumbnailConfigNames as $thumbnailConfigName) {
-                        $image->clearThumbnail($thumbnailConfigName);
-                    }
-                }
+        $thumbnail = $item['thumbnail'] ?? null;
+        if(!$thumbnail instanceof Asset\Image\Thumbnail\Config) {
+            return;
+        }
 
-                foreach ($thumbnailsToGenerate as $thumbnailConfig) {
-                    $thumbnail = $image->getThumbnail($thumbnailConfig);
+        if (isset($item['clear_thumbnail'])) {
+            $image->clearThumbnail($thumbnail->getName());
+        }
 
-                    $progress->setMessage(
-                        sprintf(
-                            'generated thumbnail for image [%d] | file: %s',
-                            $image->getId(),
-                            $thumbnail->getPath(false)
-                        )
-                    );
+        $thumbnail = $image->getThumbnail($thumbnail);
 
-                    $progress->advance(1);
-                }
-            }
-            \Pimcore::collectGarbage();
+
+        if ($output->isVerbose()) {
+            $output->writeln(
+                sprintf(
+                    'generated thumbnail for image [%d] | file: %s',
+                    $image->getId(),
+                    $thumbnail->getPath(false)
+                )
+            );
+        } else {
+            $thumbnail->getPath(false);
         }
+    }
 
-        $progress->finish();
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function runAfterBatch(InputInterface $input, OutputInterface $output)
+    {
+        \Pimcore::collectGarbage();
+    }
 
-        $output->writeln('');
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function getItemName(int $count): string
+    {
+        return 1 === $count ? 'thumbnail' : 'thumbnails';
+    }
 
-        return 0;
+    /**
+     * @return \Symfony\Component\DependencyInjection\ContainerInterface
+     */
+    protected function getContainer()
```

Comment:
please use `ContainerAwareCommand` or `ContainerAwareInterface` with `ContainerAwareTrait`, that would be a bit more Symfony like.

Reply:
done. thanks for review :)